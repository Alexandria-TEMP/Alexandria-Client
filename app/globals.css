@tailwind base;
@tailwind components;
@tailwind utilities;

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}

@layer base {
  h1 {
    @apply text-2xl font-bold;
  }
  h2 {
    @apply text-xl font-semibold;
  }
  h3 {
    @apply text-lg;
  }
  h6 {
    @apply text-sm;
  }
}
