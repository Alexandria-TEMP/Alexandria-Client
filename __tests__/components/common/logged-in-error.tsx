import { expect, describe, it } from "@jest/globals";
import NotLoggedInError from "@/components/common/logged-in-error";
import { render, screen } from "@testing-library/react";

describe("NotLoggedInError", () => {
  it("renders its children", () => {
    const { container } = render(<NotLoggedInError />);
    expect(container).toMatchSnapshot();
  });
});
