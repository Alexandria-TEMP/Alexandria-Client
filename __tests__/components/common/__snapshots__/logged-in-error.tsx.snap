// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotLoggedInError renders its children 1`] = `
<div>
  <div
    class="h-full flex flex-col justify-center items-center bg-warning-100 rounded-lg space-y-4"
    data-testid="default-error"
  >
    <h2>
      You must be logged in to post, comment or review.
    </h2>
    <div
      class="flex flex-row space-x-4"
    >
      <a
        href="/login"
      >
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
          type="button"
        >
          Log in
        </button>
      </a>
      <a
        href="/signup"
      >
        <button
          class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] data-[pressed=true]:scale-[0.97] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
          type="button"
        >
          Sign up
        </button>
      </a>
    </div>
  </div>
</div>
`;
